qemu_x86: upgrading to q35 breaks networking samples.

**Describe the bug**
with this commit, samples/net/socket/echo_server fails to run
```
commit a2aa462f7d099bf30c1e054a1f38d34359d83418 (refs/bisect/bad)
Author: Andrei Emeltchenko <andrei.emeltchenko@intel.com>
Date:   Thu Oct 27 18:54:13 2022 +0300

    qemu_x86: Use Qemu Q35 machine feature (ICH9)

    Using very old machine does not make sense anymore. Switch to the new
    Q35 (https://wiki.qemu.org/Features/Q35). Among other things it allows
    to test SMBus and watchdog.

    Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

diff --git a/boards/x86/qemu_x86/board.cmake b/boards/x86/qemu_x86/board.cmake
index b4f7c03639..934b078154 100644
--- a/boards/x86/qemu_x86/board.cmake
+++ b/boards/x86/qemu_x86/board.cmake
@@ -60,6 +60,7 @@ endif()
 set(QEMU_FLAGS_${ARCH}
   -m ${QEMU_MEMORY_SIZE_MB}
   -cpu ${QEMU_CPU_TYPE_${ARCH}}${QEMU_CPU_FLAGS}
+  -machine q35
   -device isa-debug-exit,iobase=0xf4,iosize=0x04
   ${REBOOT_FLAG}
   -nographic
```

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. west build -b qemu_x86 -- -DCONF_FILE=prj.conf -DOVERLAY_CONFIG="overlay-e1000.conf"
3. west build -t run
4. See error
```
-- west build: running target run
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: qemu32,+nx,+pae
SeaBIOS (version zephyr-v1.0.0-0-g31d4e0e-dirty-20200714_234759-fv-az50-zephyr)


iPXE (http://ipxe.org) 00:02.0 CA00 PCI2.10 PnP PMM+00392120+002F2120 CA00



Booting from ROM..
FAILED: zephyr/CMakeFiles/run_qemu /home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/net/sockets/echo_server/build/zephyr/CMakeFiles/run_qemu
cd /home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/net/sockets/echo_server/build && /home/ubuntu/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/qemu-system-i386 -m 4 -cpu qemu32,+nx,+pae -machine q35 -device isa-debug-exit,iobase=0xf4,iosize=0x04 -no-reboot -nographic -no-acpi -nic tap,model=e1000,script=no,downscript=no,ifname=zeth -pidfile qemu.pid -chardev stdio,id=con,mux=on -serial chardev:con -mon chardev=con,mode=readline -kernel /home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/net/sockets/echo_server/build/zephyr/zephyr.elf
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/net/sockets/echo_server/build --target run

```


**Expected behavior**
the qemu_x86 can start successfully

**Impact**
showstopper

**Environment (please complete the following information):**
 - OS: (e.g. Linux, , )
 - Toolchain (e.g Zephyr SDK, )
 - Commit SHA or Version used: a2aa462f7d099bf30c1e054a1f38d34359d83418 

**Additional context**
this impact the maxpro networking tcpip testing, as we testing tcpip stack on qemu_x86 platforms

--- a/boards/x86/qemu_x86/qemu_x86.dts
+++ b/boards/x86/qemu_x86/qemu_x86.dts
@@ -66,12 +66,9 @@
 			};
 		};
 
-	};
-
-	soc {
-		eth0: eth@febc0000 {
+		eth0: eth@1000 {
 			compatible = "intel,e1000";
-			reg = <0xfebc0000 0x100>;
+			reg = <PCIE_BDF(0, 2, 0) PCIE_ID(0x8086, 0x100e)>;
 			interrupts = <11 IRQ_TYPE_LOWEST_EDGE_RISING 3>;
 			interrupt-parent = <&intc>;
 

--- a/drivers/ethernet/eth_e1000.c
+++ b/drivers/ethernet/eth_e1000.c
@@ -286,6 +286,9 @@ int e1000_probe(const struct device *ddev)
 	return 0;
 }
 
+BUILD_ASSERT(DT_INST_IRQN(0) != PCIE_IRQ_DETECT,
+	     "Dynamic IRQ allocation is not supported");
+
 static void e1000_iface_init(struct net_if *iface)
 {
 	struct e1000_dev *dev = net_if_get_device(iface)->data;

--- a/drivers/ethernet/eth_e1000.c
+++ b/drivers/ethernet/eth_e1000.c
@@ -230,18 +230,16 @@ static void e1000_isr(const struct device *ddev)
 	}
 }
 
-#define PCI_VENDOR_ID_INTEL	0x8086
-#define PCI_DEVICE_ID_I82540EM	0x100e
 
 int e1000_probe(const struct device *ddev)
 {
-	const pcie_bdf_t bdf = PCIE_BDF(0, 2, 0);
+	/* PCI ID is decoded into REG_SIZE */
+	pcie_bdf_t bdf = pcie_bdf_lookup(DT_INST_REG_SIZE(0));
 	struct e1000_dev *dev = ddev->data;
 	uint32_t ral, rah;
 	struct pcie_bar mbar;
 
-	if (!pcie_probe(bdf, PCIE_ID(PCI_VENDOR_ID_INTEL,
-				     PCI_DEVICE_ID_I82540EM))) {
+	if (bdf == PCIE_BDF_NONE) {
 		return -ENODEV;
 	}
 

--- a/drivers/ethernet/eth_e1000.c
+++ b/drivers/ethernet/eth_e1000.c
@@ -235,7 +235,7 @@ static void e1000_isr(const struct device *ddev)
 
 int e1000_probe(const struct device *ddev)
 {
-	const pcie_bdf_t bdf = PCIE_BDF(0, 3, 0);
+	const pcie_bdf_t bdf = PCIE_BDF(0, 2, 0);
 	struct e1000_dev *dev = ddev->data;
 	uint32_t ral, rah;
 	struct pcie_bar mbar;

